lib/src/public_api_parser.dart

```diff
@@ -36,10 +36,26 @@ class InsertionOrderParser {
       key: (v) => v.name,
       value: (v) => v);
 
-  /// Parses a json string to [InsertionOrder].
-  static InsertionOrder parse(String jsonString) {
+  /// Parses a list of [InsertionOrder] from a [jsonString].
+  ///
+  /// Returns an empty list if [jsonString] is null or empty.
+  static List<InsertionOrder> parseInsertionOrders(String jsonString) {
+    if (jsonString == null || jsonString.isEmpty) return [];
```

Comment:
kind of concerned about the possibility of null or empty values popping up in any method. If a response was invalid i think we should fail and throw rather than propagate a null or empty string. it'll also make your life easier i think being confident in your inputs.

Reply:
I think in real cases that jsonString can never be null or empty, since even if a request fails, it still returns a json object with the error message. I'm thinking that we should add parsing to the error json and display the error to users to let them know that they have entered the wrong id, etc. I guess we should do this for all queries. 

i will create an issue and address this at end when all the queries are set. 

And the sanity check is actually just for component testing, since mock class will pass in null to this function.

Reply:
You can set the mock class to pass in a default value in setUp. i think it's better to facilitate testing needs in the test code rather than your business code

Reply:
wanted to reply to this as part of a review to draw your attention to it

Reply:
Ohh right. I will just get rid of the sanity checks then. The TODO for adding parsing support to error is at the top of the file.